src/mame/motorola/mex68kecb.cpp

```diff
@@ -0,0 +1,228 @@
+ *
+ * Created on: August 31, 2024
+ *     Author: Chris Hanson
+ *
+ * Documentation:
+ *   http://www.bitsavers.org/components/motorola/68000/MEX68KECB/MEX68KECB_D2_EduCompBd_Jul82.pdf
+ *
+ * The Motorola MC68000 Educational Computer Board is a single-board computer with
+ * a 4MHz 68000 CPU, 32KB RAM, 16KB ROM, host and terminal serial ports, a
+ * parallel interface/timer, a cassette interface, and a prototyping area with
+ * full access to the 68000 bus. The ROM contains TUTOR, a debug and bootstrap
+ * system that was the predecessor of MACSBUG.
+ *
+ * Specifications:
+ * - 4MHz MC68000L4 CPU
+ * - MC6850 ACIA x 2
+ * - MC68230 PIT
+ *
+ * To Do:
+ * - Cassette I/O
+ * - Save/Restore
+ *
+ */
+
+#include "emu.h"
+
+#include "bus/rs232/rs232.h"
+#include "cpu/m68000/m68000.h"
+#include "machine/68230pit.h"
+#include "machine/6850acia.h"
+#include "machine/mc14411.h"
+
+
+class mex68kecb_state : public driver_device
+{
+public:
+	mex68kecb_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_maincpu(*this, "maincpu")
+		, m_pit(*this, "pit")
+		, m_brg(*this, "brg")
+		, m_acia1(*this, "acia1")
+		, m_acia2(*this, "acia2")
+		, m_acia1_baud(*this, "ACIA1_BAUD")
+		, m_acia2_baud(*this, "ACIA2_BAUD")
+		, m_terminal(*this, "terminal")
+		, m_host(*this, "host")
+	{ }
+
+	void mex68kecb(machine_config &config);
+
+private:
+	required_device<cpu_device> m_maincpu;
+	required_device<pit68230_device> m_pit;
+	required_device<mc14411_device> m_brg;
+	required_device<acia6850_device> m_acia1;
+	required_device<acia6850_device> m_acia2;
+	required_ioport m_acia1_baud;
+	required_ioport m_acia2_baud;
+
+	required_device<rs232_port_device> m_terminal;
+	required_device<rs232_port_device> m_host;
+
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+	void mem_map(address_map &map);
+
+	// Clocks from Baud Rate Generator
+	template <u8 bit> void write_acia_clock(int state);
+
+	// Pointer to System ROMs needed by bootvect_r and masking RAM buffer for post reset accesses
+	uint16_t *m_sysrom = nullptr;
+	uint16_t m_sysram[8]{};
+	uint16_t bootvect_r(offs_t offset);
+	void bootvect_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+};
+
+/* Input ports */
+static INPUT_PORTS_START( mex68kecb )
+	PORT_START("ACIA1_BAUD")
+	PORT_DIPNAME(0xff, 0x80, "Terminal Baud Rate")
+	PORT_DIPSETTING(0x80, "9600") PORT_DIPLOCATION("J10:8,7,6,5,4,3,2,1")
+	PORT_DIPSETTING(0x40, "4800") PORT_DIPLOCATION("J10:8,7,6,5,4,3,2,1")
+	PORT_DIPSETTING(0x20, "2400") PORT_DIPLOCATION("J10:8,7,6,5,4,3,2,1")
+	PORT_DIPSETTING(0x10, "1200") PORT_DIPLOCATION("J10:8,7,6,5,4,3,2,1")
+	PORT_DIPSETTING(0x08,  "600") PORT_DIPLOCATION("J10:8,7,6,5,4,3,2,1")
+	PORT_DIPSETTING(0x04,  "300") PORT_DIPLOCATION("J10:8,7,6,5,4,3,2,1")
+	PORT_DIPSETTING(0x02,  "150") PORT_DIPLOCATION("J10:8,7,6,5,4,3,2,1")
+	PORT_DIPSETTING(0x01,  "110") PORT_DIPLOCATION("J10:8,7,6,5,4,3,2,1")
```

Comment:
You’re supposed to put the location on the `PORT_DIPNAME` line so you aren’t repeating it for every value.